allennlp/data/iterators/lazy_iterator.py

```diff
@@ -0,0 +1,54 @@
+import itertools
+import logging
+import math
+from typing import List
+
+from overrides import overrides
+
+from allennlp.common import Params
+from allennlp.data.iterators.data_iterator import DataIterator
+from allennlp.data.dataset import Dataset
+from allennlp.data.instance import Instance
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@DataIterator.register("lazy")
+class LazyIterator(DataIterator):
+    """
+    An iterator for iterating lazily across a dataset, yielding fixed size batches.
+
+    Parameters
+    ----------
+    batch_size : int, optional, (default = 32)
+        The size of each batch of instances yielded when calling the iterator.
+    """
+    def __init__(self, batch_size: int = 32) -> None:
+        self._batch_size = batch_size
+
+    def _yield_one_epoch(self, dataset: Dataset, shuffle: bool, cuda_device: int, for_training: bool):
+        if shuffle:
+            logger.warning("shuffle is not implemented for LazyIterator")
```

Comment:
This seems liable to create bugs in the future, since we might use this private method in some generic way that requires shuffle and not notice the message in the logs. If this is a private method perhaps it should throw an error instead?

Reply:
Yeah, a `ConfigurationError` does seem warranted here, as this would be the result of mis-configuring something in your json file.